### PR TITLE
feat: Add the Wave Sort algorithm

### PR DESCRIPTION
**Pull request by Swastyy**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1525"></a>

### CODE REVIEW

File: sorting/wave_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/**
+ * @file
+ * @brief Implementation of [Wave sort] algorithm
```
Comment (Panquesito7) [inline sorting/wave_sort.cpp:3]: Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation.

File: sorting/wave_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * @brief Sorting algorithms
+ */
+namespace sorting {
+/**
+ * @namespace wave_sort
+ * @brief Functions for [Wave sort] algorithm
+ */
+namespace wave_sort {
+/**
+ * @brief The main function implements waveSort
```
Comment (Panquesito7) [inline sorting/wave_sort.cpp:26]:
```suggestion
 * @brief The main function implements that implements the Wave Sort algorithm
```

File: sorting/wave_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    for (int64_t i = 0; i < n - 1; i += 2) {  // swap all the adjacent elements
+        std::swap(arr[i], arr[i + 1]);
+    }
+    return arr;
+}
+}  // namespace wave_sort
+}  // namespace sorting
+
+/**
+ * @brief Test implementations
```
Comment (Panquesito7) [inline sorting/wave_sort.cpp:48]:
```suggestion
 * @brief Self-test implementations
```

File: sorting/wave_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    assert(arr4 == o4);
+    std::cout << "passed" << std::endl;
+}
+
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();  // execute the test
```
Comment (Panquesito7) [inline sorting/wave_sort.cpp:90]:
```suggestion
    test();  // run self-test implementations
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 👍

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
It's just to accept these suggestions, and you're done! 😄

**Review by Panquesito7 — APPROVED** [review]
LGTM. Great work, @Swastyy! Thank you for your contribution! 👍 🎉

**Review by ayaankhan98 — APPROVED** [review]

**Review by mishraabhinn — APPROVED** [review]

**Pull request merged**